crates/matrix-sdk/src/room/mod.rs

```diff
@@ -1512,7 +1512,23 @@ impl Room {
 
                 if !self.are_members_synced() {
                     self.sync_members().await?;
-                    // TODO query keys here?
+
+                    // Query keys in case we don't have them for newly synced members.
+                    let olm = self.client.olm_machine().await;
+                    let olm = olm.as_ref().expect("Olm machine wasn't started");
+
+                    let members = self
```

Comment:
One thing that sucks here is that SS doesn't support delta tokens, meaning `self.are_members_synced()` returns `false` very often (I forgot the exact case when we reset the flag, every time we re-subscribe to the room?) instead of only once like in the sync V2 case.

We could perhaps optimize this if we filter out members that are marked to be tracked and not considered to be dirty.